modules/test/playwright/tests/site-cms-site-initializer/main/recycleBin.spec.ts

```diff
@@ -144,3 +144,131 @@ test(
 		});
 	}
 );
+
+test(
+	'Can restore a folder and its contents from Recycle Bin',
+	{tag: '@LPD-59716'},
+	async ({contentsPage, folderPage, page, recycleBinPage}) => {
+		test.slow();
+
+		const folderName = getRandomString();
+		const nestedFolderName = `nested-${getRandomString()}`;
+		const contentName = getRandomString();
+		const friendlyUrl = getRandomString();
+
+		await test.step('Create a folder with nested content', async () => {
+			await contentsPage.goto();
+
+			await folderPage.createFolder(folderName);
+
+			await page
+				.getByRole('row', {name: folderName})
+				.getByRole('link')
+				.click();
+
+			await folderPage.createFolder(nestedFolderName);
+
+			await page
+				.getByRole('row', {name: nestedFolderName})
+				.getByRole('link')
+				.click();
+
+			await contentsPage.createContent('Knowledge Base');
```

Comment:
To have a faster test execution, we should use the API directly instead of page navigations to operate with items when this navigation is not part of the tests. In this case we can use:

```typescript
		const folderData =
			await apiHelpers.objectFolder.createObjectEntryFolder({
				scopeKey: 'Default',
				title: folderTitle,
			});

		const applicationName = 'cms/basic-documents';
		const objectEntry = await apiHelpers.objectEntry.postObjectEntry(
			{
				file: {
					fileBase64: 'R0lGODlhAQABAAAAACw=',
					name: `file_${getRandomString()}.png`,
				},
				objectEntryFolderExternalReferenceCode: 'L_FILES',
				title: `title ${getRandomString()}`,
			},
			applicationName,
			'Default'
		);
```